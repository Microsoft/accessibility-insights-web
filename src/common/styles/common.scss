// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
@import 'high-contrast-override.scss';
@import 'colors.scss';
@import 'fonts.scss';

$focus-ring-color: $neutral-100;

@mixin set-focus-ring-styles($property: 'border') {
    #{$property}: 1px solid $focus-ring-color;
    @include ms-high-contrast-override {
        #{$property}: 1px dashed windowtext !important;
    }
}

.ms-Fabric.is-focusVisible .ms-Nav-group .ms-nav-linkButton:focus::after {
    @include set-focus-ring-styles('border');
}

button::after {
    content: '';
    position: absolute;
}

.ms-Button.ms-Button--primary:not(.is-disabled) {
    background: $communication-primary;

    &:hover {
        background: $communication-shade-10;
    }
}

.ms-fontColor-neutralPrimary,
.ms-fontColor-neutralPrimary--hover:hover {
    color: $neutral-100 !important;
}

.ms-Fabric {
    color: $neutral-100;
}

.generic-toggle-component {
    margin-bottom: 4vh;
    .toggle-container {
        margin-bottom: 1vh;
        .toggle-name {
            display: inline-block;
            font-size: 17px;
            font-weight: bold;
        }

        .toggle {
            float: right;
        }
    }
}

.high-contrast-theme {
    .insights-link {
        &:hover {
            text-decoration: underline;
        }
    }
    .is-selected {
        .status-icon {
            color: $high-contrast-icon-color !important;
        }
    }
}

.insights-link {
    // marking important to override office fabric style
    color: $link !important;
    text-decoration: none;

    &:hover {
        color: $link-hover !important;
        text-decoration: underline;
    }
}

.ms-Spinner {
    .ms-Spinner-circle {
        border-top-color: $spinner-text;
    }

    .ms-Spinner-label {
        color: $spinner-text;
    }
}

$normalTitleFontWeight: 300 !default;

$pivotItemLeftBorderWidth: 3px !default;
$pivotItemBorderStyle: solid !default;

$detailsViewHeaderBarHeight: 40px !default;
$detailsViewNavPivotsHeight: 41px !default;
$detailsViewCommandBarHeight: 40px !default;
$detailsViewCommandBarBorderHeight: 1px !default;
$detailsViewLeftNavWidth: 200px !default;
$detailsViewTotalHeaderHeight: (#{$detailsViewCommandBarBorderHeight} + #{$detailsViewCommandBarHeight} + #{$detailsViewHeaderBarHeight});

$fastPassRightPanelMarginRight: 24px !default;
$fastPassRightPanelMarginLeft: 24px !default;
$fastPassRightPanelMarginTop: 24px !default;

@mixin ellipsedText {
    white-space: nowrap;
    overflow: hidden;
    text-overflow: ellipsis;
}
