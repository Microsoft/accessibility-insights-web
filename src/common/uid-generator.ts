// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import * as uuid4 from 'uuid/v4';

export type UUIDGenerator = () => string;

export let generateUID: UUIDGenerator = uuid4;
