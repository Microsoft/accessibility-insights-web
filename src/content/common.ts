// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import * as react from 'react';

import { link } from './link';
import { ContentCreator } from '../views/content/content-page';
import { GuidanceTitle as gt } from '../views/content/guidance-title';

export const create = ContentCreator(link);
export const React = react;
export const GuidanceTitle = gt;
