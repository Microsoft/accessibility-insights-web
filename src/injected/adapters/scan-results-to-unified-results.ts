// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

import { ResultStatus, UnifiedResult, UnifiedResults } from '../../common/types/store-data/unified-data-interface';
import { AxeNodeResult, RuleResult, ScanResults } from '../../scanner/iruleresults';

type UUIDGeneratorType = () => string;

interface RuleResultData {
    status: ResultStatus;
    ruleID: string;
}

interface CreationData extends RuleResultData {
    cssSelector: string;
    snippet: string;
    howToFix: {
        oneOf: string[];
        none: string[];
        all: string[];
    };
}

export const convertScanResultsToUnifiedResults = (scanResults: ScanResults, uuidGenerator: UUIDGeneratorType): UnifiedResults => {
    return {
        results: createUnifiedResultsFromScanResults(scanResults, uuidGenerator),
    } as UnifiedResults;
};

function createUnifiedResultsFromScanResults(scanResults: ScanResults, uuidGenerator: UUIDGeneratorType): UnifiedResult[] {
    let resultsArray: UnifiedResult[] = [];

    resultsArray = createUnifiedResultsFromRuleResults(scanResults.violations, 'fail', uuidGenerator);
    resultsArray = resultsArray.concat(createUnifiedResultsFromRuleResults(scanResults.passes, 'pass', uuidGenerator));

    return resultsArray;
}

function createUnifiedResultsFromRuleResults(
    ruleResults: RuleResult[],
    status: ResultStatus,
    uuidGenerator: UUIDGeneratorType,
): UnifiedResult[] {
    let unifiedResults: UnifiedResult[] = [];

    (ruleResults || []).forEach(result => {
        const moreUnifiedResults = createUnifiedResultsFromRuleResult(result, status, uuidGenerator);
        unifiedResults = unifiedResults.concat(moreUnifiedResults);
    });

    return unifiedResults;
}

function createUnifiedResultsFromRuleResult(
    ruleResult: RuleResult,
    status: ResultStatus,
    uuidGenerator: UUIDGeneratorType,
): UnifiedResult[] {
    return ruleResult.nodes.map(node => {
        const data: RuleResultData = {
            status: status,
            ruleID: ruleResult.id,
        };

        return createUnifiedResultFromNode(node, data, uuidGenerator);
    });
}

function createUnifiedResultFromNode(
    nodeResult: AxeNodeResult,
    ruleResultData: RuleResultData,
    uuidGenerator: UUIDGeneratorType,
): UnifiedResult {
    return createUnifiedResult(
        {
            ...ruleResultData,
            cssSelector: nodeResult.target.join(';'),
            snippet: nodeResult.snippet || nodeResult.html,
            howToFix: {
                oneOf: nodeResult.any.map(checkResult => checkResult.message),
                none: nodeResult.none.map(checkResult => checkResult.message),
                all: nodeResult.all.map(checkResult => checkResult.message),
            },
        },
        uuidGenerator,
    );
}

function createUnifiedResult(data: CreationData, uuidGenerator: UUIDGeneratorType): UnifiedResult {
    return {
        uid: uuidGenerator(),
        status: data.status,
        ruleId: data.ruleID,
        identifiers: {
            'css-selector': data.cssSelector,
        },
        descriptors: {
            snippet: data.snippet,
        },
        resolution: {
            'how-to-fix-web': {
                any: data.howToFix.oneOf,
                none: data.howToFix.none,
                all: data.howToFix.all,
            },
        },
    } as UnifiedResult;
}
