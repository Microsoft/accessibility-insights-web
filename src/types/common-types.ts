// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
interface DictionaryStringTo<T> {
    [key: string]: T;
}

interface DictionaryNumberTo<T> {
    [key: number]: T;
}

interface FunctionPPR<TParam1, TParam2, TResult> {
    (param1: TParam1, param2: TParam2): TResult;
}
