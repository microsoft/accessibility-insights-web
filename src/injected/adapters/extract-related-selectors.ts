// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { TargetHelper } from 'common/target-helper';
import { uniq } from 'lodash';
import { AxeNodeResult } from '../../scanner/iruleresults';

export type RelatedSelectorExtractor = typeof extractRelatedSelectors;

export function extractRelatedSelectors(nodeResult: AxeNodeResult): string[] | undefined {
    const nodeSelector = TargetHelper.getSelectorFromTarget(nodeResult.target);

    let relatedSelectors: string[] = [];
    for (const checkType of ['all', 'any', 'none'] as const) {
        const checks = nodeResult[checkType];
        for (const check of checks) {
            const relatedSelectorsForCheck =
                check.relatedNodes?.map(n => TargetHelper.getSelectorFromTarget(n.target)) ?? [];
            relatedSelectors.push(...relatedSelectorsForCheck);
        }
    }

    // This doesn't usually change anything; it would mainly affect if a rule had multiple
    // checks that each returned overlapping sets of related nodes. Axe generally returns
    // nodes in a reasonable order, so we want to use a method that preserves order here.
    relatedSelectors = uniq(relatedSelectors);

    // This does have a practical impact; some axe checks (eg, color-contrast) will occasionally
    // return the node itself as a "related node", and we'd rather avoid this.
    relatedSelectors = relatedSelectors.filter(relatedSelector => relatedSelector !== nodeSelector);

    return relatedSelectors.length === 0 ? undefined : relatedSelectors;
}
