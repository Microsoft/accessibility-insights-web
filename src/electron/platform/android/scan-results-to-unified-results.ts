// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

import { InstancePropertyBag, InstanceResultStatus, UnifiedResult } from 'common/types/store-data/unified-data-interface';
import { UUIDGeneratorType } from 'common/uid-generator';
import { DictionaryStringTo } from '../../../types/common-types';
import { RuleInformation } from './rule-information';
import { RuleInformationProvider } from './rule-information-provider';
import { RuleResultsData, ScanResults, ViewElementData } from './scan-results';

export type ConvertScanResultsToUnifiedResultsDelegate = (scanResults: ScanResults, uuidGenerator: UUIDGeneratorType) => UnifiedResult[];

export function convertScanResultsToUnifiedResults(scanResults: ScanResults, uuidGenerator: UUIDGeneratorType): UnifiedResult[] {
    if (!scanResults || !scanResults.ruleResults) {
        return [];
    }

    return createUnifiedResultsFromScanResults(scanResults, uuidGenerator);
}

function createUnifiedResultsFromScanResults(scanResults: ScanResults, uuidGenerator: UUIDGeneratorType): UnifiedResult[] {
    const ruleInformationProvider: RuleInformationProvider = new RuleInformationProvider();
    const viewElementLookup: DictionaryStringTo<ViewElementData> = createViewElementLookup(scanResults);
    const unifiedResults: UnifiedResult[] = [];

    for (const ruleResult of scanResults.ruleResults) {
        const ruleInformation: RuleInformation = ruleInformationProvider.getRuleInformation(ruleResult.ruleId);

        if (ruleInformation) {
            unifiedResults.push(createUnifiedResult(ruleInformation, ruleResult, viewElementLookup, uuidGenerator));
        }
    }

    return unifiedResults;
}

function createViewElementLookup(scanResults: ScanResults): DictionaryStringTo<ViewElementData> {
    const viewElementLookup = {};

    addViewElementAndChildren(viewElementLookup, scanResults.viewElementTree);

    return viewElementLookup;
}

function addViewElementAndChildren(viewElementLookup: DictionaryStringTo<ViewElementData>, element: ViewElementData): void {
    if (element) {
        viewElementLookup[element.axeViewId] = element;
        if (element.children) {
            for (const child of element.children) {
                addViewElementAndChildren(viewElementLookup, child);
            }
        }
    }
}

function createUnifiedResult(
    ruleInformation: RuleInformation,
    ruleResult: RuleResultsData,
    viewElementLookup: DictionaryStringTo<ViewElementData>,
    uuidGenerator: UUIDGeneratorType,
): UnifiedResult {
    return {
        uid: uuidGenerator(),
        ruleId: ruleInformation.ruleId,
        status: getStatus(ruleResult.status),
        descriptors: getDescriptors(viewElementLookup[ruleResult.axeViewId]),
        identifiers: null,
        resolution: {
            'how-to-fix': ruleInformation.howToFix(ruleResult),
        },
    };
}

function getDescriptors(viewElement: ViewElementData): InstancePropertyBag {
    if (viewElement) {
        return {
            className: viewElement.className,
            boundingRectangle: viewElement.boundsInScreen,
            contentDescription: viewElement.contentDescription,
            text: viewElement.text,
        };
    }
    return null;
}

function getStatus(status: string): InstanceResultStatus {
    switch (status) {
        case 'PASS':
            return 'pass';
        case 'FAIL':
            return 'fail';
        default:
            return 'unknown';
    }
}
