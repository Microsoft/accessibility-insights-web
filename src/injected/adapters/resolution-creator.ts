// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

import { UnifiedResolution } from 'common/types/store-data/unified-data-interface';
import { ResolutionType } from 'injected/adapters/scan-results-to-unified-results';
import { AxeNodeResult } from '../../scanner/iruleresults';

export type ResolutionCreator = (
    resolution: ResolutionType,
    data: ResolutionCreatorData,
) => UnifiedResolution;

interface ResolutionCreatorData {
    id: string;
    nodeResult: AxeNodeResult;
}

// split into two, one for fix and one for change.  Pass both into unified result sender, change up report & mocks, main window initializer too
export const getResolution: ResolutionCreator = (
    resolution: ResolutionType,
    data: ResolutionCreatorData,
) => {
    if (resolution === 'how-to-check') {
        return {
            howToFixSummary: data.failureSummary,
            'how-to-check-web': getHowToCheckTest(data.ruleID),
        };
    } else {
        return {
            howToFixSummary: data.failureSummary,
            'how-to-fix-web': {
                any: data.howToFix.oneOf,
                none: data.howToFix.none,
                all: data.howToFix.all,
            },
        };
    }
};

const getHowToCheckTest = (ruleID: string) => {
    let checkText: string;
    switch (ruleID) {
        case 'aria-input-field-name': {
            checkText =
                "Inspect the element using the Accessibility pane in the browser Developer Tools to verify that the field's accessible name is complete without its associated <label>.";
            break;
        }
        case 'color-contrast': {
            checkText =
                "If the text is intended to be invisible, it passes.\nIf the text is intended to be visible, use Accessibility Insights for Windows (or the Colour Contrast Analyser if you're testing on a Mac) to manually verify that it has sufficient contrast compared to the background. If the background is an image or gradient, test an area where contrast appears to be lowest.\nFor detailed test instructions, see Assessment > Text legibility > Contrast.";
            break;
        }
        case 'link-in-text-block': {
            checkText =
                "Manually verify that the link text EITHER has a contrast ratio of at least 3:1 compared to surrounding text OR has a distinct visual style (such as underlined, bolded, or italicized). To measure contrast, use Accessibility Insights for Windows (or the Colour Contrast Analyser if you're testing on a Mac).";
            break;
        }
        case 'th-has-data-cells': {
            checkText =
                'Examine the header cell in the context of the table to verify that it has no data cells.';
            break;
        }
        default: {
            checkText =
                "No 'How to check' guidance has been supplied.  Please contact the Accessibility Insights team.";
            break;
        }
    }
    return checkText;
};
