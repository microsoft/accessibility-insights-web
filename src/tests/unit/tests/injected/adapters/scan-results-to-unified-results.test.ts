// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

import { RelatedSelectorExtractor } from 'injected/adapters/extract-related-selectors';
import { ResolutionCreator } from 'injected/adapters/resolution-creator';
import { ConvertScanResultsToUnifiedResults } from 'injected/adapters/scan-results-to-unified-results';
import { IMock, Mock, MockBehavior, Times } from 'typemoq';
import { DictionaryStringTo } from 'types/common-types';
import { generateUID } from '../../../../../common/uid-generator';
import { RuleResult, ScanResults } from '../../../../../scanner/iruleresults';

describe('ScanResults to Unified Results Test', () => {
    let generateGuidMock: IMock<() => string>;
    let fixResolutionCreatorMock: IMock<ResolutionCreator>;
    let checkResolutionCreatorMock: IMock<ResolutionCreator>;
    let extractRelatedSelectorsStub: RelatedSelectorExtractor;

    beforeEach(() => {
        const guidStub = 'gguid-mock-stub';
        generateGuidMock = Mock.ofInstance(generateUID, MockBehavior.Strict);
        generateGuidMock
            .setup(ggm => ggm())
            .returns(() => guidStub)
            .verifiable(Times.atLeastOnce());
        fixResolutionCreatorMock = Mock.ofType<ResolutionCreator>();
        checkResolutionCreatorMock = Mock.ofType<ResolutionCreator>();
        extractRelatedSelectorsStub = node => [`extractRelatedSelectors output for ${node.target}`];
    });

    const nullIdentifiers = [null, undefined, {}];

    const fixResolutionStub: DictionaryStringTo<any> = { 'test fix resolution': {} };
    const checkResolutionStub: DictionaryStringTo<any> = { 'test check resolution': {} };

    test.each(nullIdentifiers)(
        'automatedChecksConversion provides a defined UnifiedResult instance %s',
        scanResultStub => {
            const testSubject = getTestSubject();
            const unifiedResults = testSubject.automatedChecksConversion(
                scanResultStub as ScanResults,
            );
            expect(unifiedResults).toBeDefined();
        },
    );

    test('automaticChecksConversion works fine when there is no data in scanresults', () => {
        const testSubject = getTestSubject();
        const scanResultsStub: ScanResults = createTestResultsWithNoData();
        expect(testSubject.automatedChecksConversion(scanResultsStub)).toMatchSnapshot();
    });

    test('automaticChecksConversion works with filled up passes and failures value in scan results', () => {
        setupResolutionMock(fixResolutionCreatorMock, 'test', passingNode, fixResolutionStub);
        setupResolutionMock(fixResolutionCreatorMock, 'id1', node1, fixResolutionStub);
        setupResolutionMock(fixResolutionCreatorMock, 'id1', node2, fixResolutionStub);
        setupResolutionMock(fixResolutionCreatorMock, 'id2', node3, fixResolutionStub);

        const testSubject = getTestSubject();
        const scanResultsStub: ScanResults = createTestResults();
        expect(testSubject.automatedChecksConversion(scanResultsStub)).toMatchSnapshot();
        generateGuidMock.verifyAll();
    });

    test.each(nullIdentifiers)(
        'needsReviewConversion provides a defined UnifiedResult instance %s',
        scanResultStub => {
            const testSubject = getTestSubject();
            const unifiedResults = testSubject.needsReviewConversion(scanResultStub as ScanResults);
            expect(unifiedResults).toBeDefined();
        },
    );

    test('needsReviewConversion works fine when there is no data in scanresults', () => {
        const testSubject = getTestSubject();
        const scanResultsStub: ScanResults = createTestResultsWithNoData();
        expect(testSubject.needsReviewConversion(scanResultsStub)).toMatchSnapshot();
    });

    test('needsReviewConversion works with filled up passes, failures and incomplete values in scan results', () => {
        setupResolutionMock(checkResolutionCreatorMock, 'test', passingNode, checkResolutionStub);
        setupResolutionMock(checkResolutionCreatorMock, 'id1', node1, checkResolutionStub);
        setupResolutionMock(checkResolutionCreatorMock, 'id1', node2, checkResolutionStub);
        setupResolutionMock(checkResolutionCreatorMock, 'id2', node3, checkResolutionStub);
        setupResolutionMock(
            checkResolutionCreatorMock,
            'test2',
            incompleteNode,
            checkResolutionStub,
        );

        const testSubject = getTestSubject();
        const scanResultsStub: ScanResults = createTestResultsWithIncompletes();
        expect(testSubject.needsReviewConversion(scanResultsStub)).toMatchSnapshot();
        generateGuidMock.verifyAll();
    });

    function setupResolutionMock(
        mock: IMock<ResolutionCreator>,
        id: string,
        nodeResult: AxeNodeResult,
        resolutionStub: DictionaryStringTo<any>,
    ): void {
        mock.setup(m =>
            m({
                ruleId: id,
                nodeResult: nodeResult,
            }),
        ).returns(() => resolutionStub);
    }

    function getTestSubject(): ConvertScanResultsToUnifiedResults {
        return new ConvertScanResultsToUnifiedResults(
            generateGuidMock.object,
            fixResolutionCreatorMock.object,
            checkResolutionCreatorMock.object,
            extractRelatedSelectorsStub,
        );
    }

    function createTestResultsWithNoData(): ScanResults {
        return {
            passes: [],
            violations: [],
            incomplete: [],
            targetPageTitle: '',
            targetPageUrl: '',
        } as ScanResults;
    }

    const node1: AxeNodeResult = {
        any: [
            {
                id: 'test-id',
                message: 'any-test-message',
                data: {},
            },
        ],
        none: [],
        all: [],
        instanceId: 'id1',
        html: 'html1',
        target: ['target1', 'id1'],
        failureSummary: 'how to fix 1',
    };

    const node2: AxeNodeResult = {
        any: [],
        none: [
            {
                id: 'none-test-id',
                message: 'none-test-message',
                data: {},
            },
        ],
        all: [],
        instanceId: 'id2',
        snippet: 'html2',
        target: ['target2', 'id2'],
        failureSummary: 'how to fix 2',
    } as AxeNodeResult;

    const node3: AxeNodeResult = {
        any: [],
        none: [],
        all: [
            {
                id: 'all-test-id',
                message: 'all-test-message',
                data: {},
            },
        ],
        instanceId: 'id3',
        html: 'html3',
        target: ['target3', 'id3'],
        failureSummary: 'how to fix 3',
    };

    const passingNode: AxeNodeResult = {
        any: [],
        none: [],
        all: [],
        instanceId: 'id-pass',
        html: 'html-pass',
        target: ['passTarget1', 'passTarget2'],
    };

    const incompleteNode: AxeNodeResult = {
        any: [],
        none: [],
        all: [],
        instanceId: 'id-incomplete',
        html: 'html-incomplete',
        target: ['incompleteTarget1'],
    };

    const failedRules: RuleResult[] = [
        {
            id: 'id1',
            nodes: [node1, node2],
            description: 'des1',
            help: 'help1',
            guidanceLinks: [
                {
                    text: 'text1',
                    href: 'testurl1',
                },
            ],
            helpUrl: 'https://id1',
        },
        {
            id: 'id2',
            nodes: [node3],
            description: 'des2',
            help: 'help2',
            guidanceLinks: [
                {
                    text: 'text2',
                    href: 'testurl2',
                },
            ],
            helpUrl: 'https://id2',
        },
    ];

    function createTestResults(): ScanResults {
        return {
            passes: [
                {
                    id: 'test',
                    nodes: [passingNode],
                },
            ],
            violations: failedRules,
            targetPageTitle: '',
            targetPageUrl: '',
        } as ScanResults;
    }

    function createTestResultsWithIncompletes(): ScanResults {
        return {
            passes: [
                {
                    id: 'test',
                    nodes: [passingNode],
                },
            ],
            violations: failedRules,
            incomplete: [
                {
                    id: 'test2',
                    nodes: [incompleteNode],
                },
            ],
            targetPageTitle: '',
            targetPageUrl: '',
        } as ScanResults;
    }
});
