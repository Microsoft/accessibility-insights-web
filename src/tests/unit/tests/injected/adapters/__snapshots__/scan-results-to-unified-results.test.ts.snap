// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScanResults to Unified Results Test conversion works with filled up passes and failures value in scanresults 1`] = `
Object {
  "results": Array [
    Object {
      "descriptors": Object {
        "snippet": "html1",
      },
      "identifiers": Object {
        "css-selector": "target1;id1",
      },
      "resolution": Object {
        "how-to-fix-web": Object {
          "all": Array [],
          "any": Array [],
          "none": Array [],
        },
      },
      "ruleId": "id1",
      "status": "fail",
      "uid": "gguid-mock-stub",
    },
    Object {
      "descriptors": Object {
        "snippet": "html2",
      },
      "identifiers": Object {
        "css-selector": "target2;id2",
      },
      "resolution": Object {
        "how-to-fix-web": Object {
          "all": Array [],
          "any": Array [],
          "none": Array [],
        },
      },
      "ruleId": "id1",
      "status": "fail",
      "uid": "gguid-mock-stub",
    },
    Object {
      "descriptors": Object {
        "snippet": "html3",
      },
      "identifiers": Object {
        "css-selector": "target3;id3",
      },
      "resolution": Object {
        "how-to-fix-web": Object {
          "all": Array [],
          "any": Array [],
          "none": Array [],
        },
      },
      "ruleId": "id2",
      "status": "fail",
      "uid": "gguid-mock-stub",
    },
    Object {
      "descriptors": Object {
        "snippet": "html-pass",
      },
      "identifiers": Object {
        "css-selector": "passTarget1;passTarget2",
      },
      "resolution": Object {
        "how-to-fix-web": Object {
          "all": Array [],
          "any": Array [],
          "none": Array [],
        },
      },
      "ruleId": "test",
      "status": "pass",
      "uid": "gguid-mock-stub",
    },
  ],
}
`;

exports[`ScanResults to Unified Results Test convesion works fine when the only value in the scanresults is timeStamp 1`] = `
Object {
  "results": Array [],
}
`;
